src/data/mod.rs

```diff
@@ -581,18 +582,17 @@ mod tests {
             "struct s",
         ];
         for ty in types.iter() {
+            let tokens = Lexer::new("<integration-test>", ty.chars(), false).collect::<Vec<_>>();
+
             assert_eq!(
                 &format!(
                     "{}",
-                    crate::Parser::new(
-                        crate::Lexer::new("<integration-test>", ty.chars(), false),
-                        false
-                    )
-                    .unwrap()
-                    .type_name()
-                    .unwrap()
-                    .data
-                    .0
+                    Parser::new(tokens.into_iter(), false)
```

Comment:
Could you change this back so it takes a lexer directly? Or at least remove the collect()? I know this is nitpicky.